### PR TITLE
[Experimental] Allow disabling product descriptions in the order summary.

### LINKED ISSUE #51886: [Enhancement]: Remove product descriptions from the Cart/Checkout blocks but add a way to re-enable them for 3PDs
### Describe the solution you'd like

In https://github.com/woocommerce/woocommerce/pull/45767 we removed the product description from items in the Cart/Checkout blocks.

We reverted this change as we found out 3PDs were using this for reasons beyond describing the product, this was breaking their integration.

The solution to this is to remove the product description by default, but enable a way for 3PDs to turn it back on.

Ideally we will ship this as an experimental feature (he description remains enabled unless the experimental feature is turned on, which will disable the description) This gives 3PDs time to update their code and in 2-3 versions time we will graduate this to a stable feature, meaning descriptions will be turned _off_ by default.

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by samueljseay**
### Changes proposed in this Pull Request:

Closes #51886

Introduce's a new block setting to the order summary items. It is disabled by default and also not available to enable unless you have the `experimental-blocks` flag enabled.

When the block setting toggle is enabled we do not show product descriptions in the order summary.

### How to test the changes in this Pull Request:

0. Enable experimental-blocks feature flag (should be enabled in dev by default), Have some products with a description
1. Now go to checkout page in editor
2. Within the block hierarchy select order summary items block (within order summary)
3. Note there should be a toggle available "Disable product descriptions"
4. Enable the toggle, see in the editor that product descriptions in the preview disappear
5. Save changes and view the front-end as a shopper and note that when you go to checkout your products do not show product descriptions.
6. Go back to editor and disable the toggle again and see that product descriptions are displayed (and in the frontend)

Before  | After
------------- | -------------
Screenshot 2024-11-01 at 2 15 10 PM  | Screenshot 2024-11-01 at 2 15 41 PM

### Changelog entry

-   [x] Automatically create a changelog entry from the details below.

-   [ ] This Pull Request does not require a changelog entry. (Comment required below)

<details>

<summary>Changelog Entry Details</summary>

#### Significance

-   [ ] Patch
-   [x] Minor
-   [ ] Major

#### Type

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [x] Enhancement - Improvement to existing functionality

#### Message 
[Experimental] Allow disabling product descriptions in the product summary.

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>

### CODE REVIEW

File: plugins/woocommerce/package.json [hunk 1]
```diff
@@ -786,8 +786,10 @@
 				"node_modules/@woocommerce/e2e-core-tests/CHANGELOG.md",
 				"node_modules/@woocommerce/api/dist/",
 				"node_modules/@woocommerce/admin-e2e-tests/build",
+				"node_modules/@woocommerce/classic-assets/build",
 				"node_modules/@woocommerce/block-library/build",
 				"node_modules/@woocommerce/block-library/blocks.ini",
+				"node_modules/@woocommerce/admin-library/build",
```
Comment (wavvves) [inline plugins/woocommerce/package.json]: Are these changes required? I get the same changes always after pnpm install

**Review by wavvves — APPROVED** [review]
Tested out ok, but left a comment, there are some package.json changes that should probably be reverted 👍🏼

File: plugins/woocommerce/package.json (hunk 1 above)
Comment (samueljseay) [reply plugins/woocommerce/package.json]: Ah yeah, I will revert this, thanks 🙇

Comment (samueljseay) [reply plugins/woocommerce/package.json]: Looks like this change has already been reverted.
